eos-image-boot-dm-setup

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+#
+# When booting from an image file hosted on a filesystem, this program maps the
+# host filesystem to a dm-device with a hole where the image file lives.
+
+for i in $(</proc/cmdline); do
+  case $i in
+  endless.image.device=*)
+    host_device=${i//endless.image.device=}
+    ;;
+  endless.image.path=*)
+    image_path=${i//endless.image.path=}
+    ;;
+  esac
+done
+
+dm_roflag="--readonly"
+
+[ -z "${host_device}" -o -z "${image_path}" ] && exit 1
+
+case "${host_device}" in
+PARTUUID=*)
+  host_device=/dev/disk/by-partuuid/${host_device#PARTUUID=}
+  ;;
+UUID=*)
+  host_device=/dev/disk/by-uuid/${host_device#UUID=}
+  ;;
+esac
+
+host_device_size=$(blockdev --getsz ${host_device})
+if [ $? != 0 ]; then
+  echo "Failed to read ${host_device} size"
+  exit 1
+fi
+
+fstype=$(blkid -o value -s TYPE "${host_device}")
+if [ $? != 0 ]; then
+  echo "Failed to detect filesystem type on ${host_device}"
+  exit 1
+fi
+
+case "${fstype}" in
+exfat)
+  extents=$(dumpexfat -f "${image_path}" "${host_device}" | sort -n)
+  ;;
+ntfs)
+  extents=$(ntfsextents "${host_device}" "${image_path}" | sort -n)
```

Comment:
Unfortunately I think this now masks errors from `ntfsextents` because the return code of a pipeline is the return code of the rightmost command (unless you `set -o pipefail`):

```bash
  tensionsheet ~
% true | false; echo $?
1
  tensionsheet ~
% false | true; echo $?
0
  tensionsheet ~
% set -o pipefail; false | true; echo $?
1
```